### PR TITLE
NOTICKET constructor.name mangling

### PR DESCRIPTION
**Pull request by FostUK**
* Fixes problem with webpack mangling constructor.name (should not rely on constructor.name and use constructor instead)
* Fix demo game tests that were relying on a class property collision between .list (now scrollableList) and Phaser's .list

### CODE REVIEW

File: src/components/shop/shop-menu-screen.js [hunk 1]
```diff
@@ -10,10 +10,12 @@ import { onScaleChange } from "../../core/scaler.js";
 import { createTitles } from "../../core/titles.js";
 import { createBalance } from "./balance-ui.js";
 import { createMenu } from "./menu.js";
+import { initResizers } from "./backgrounds.js";
 
 export class ShopMenu extends Screen {
     preload() {
         this.plugins.installScenePlugin("rexUI", RexUIPlugin, "rexUI", this, true);
+        initResizers();
```
Comment (AnthonyGoodchild) [inline src/components/shop/shop-menu-screen.js:18]: can you init the resizers as part of `addBackgroundItems`, or would that be polluting it with Rex?

**Review by AnthonyGoodchild — APPROVED** [review]
good fix 👍

File: src/components/shop/shop-list-screen.js [hunk 2]
```diff
@@ -11,18 +11,15 @@ import { onScaleChange } from "../../core/scaler.js";
...
         this.plugins.installScenePlugin("rexUI", RexUIPlugin, "rexUI", this, true);
+        initResizers();
     }
 
     create() {
-        if (this.transientData.shop.mode === "shop") {
-            gmi.sendStatsEvent("shopbuy", "click", {});
-        } else {
-            gmi.sendStatsEvent("shopmanage", "click", {});
-        }
```
Comment (AdamXA) [inline src/components/shop/shop-list-screen.js:25]: we need this stat right?

Comment (AdamXA) [reply src/components/shop/shop-list-screen.js:25]: hm did this not fail a unit test either

Comment (FostUK) [reply src/components/shop/shop-list-screen.js:25]: That's kind of strange. I never touched that. Will revert and check.

Comment (FostUK) [reply src/components/shop/shop-list-screen.js:25]: Added back in but made it a one-liner.

File: src/components/shop/shop-menu-screen.js (hunk 1 above)
Comment (FostUK) [reply src/components/shop/shop-menu-screen.js:18]: Sadly that is called for every scene - not just ones with the ui plugin included.

**Pull request merged**